numpy/core/tests/test_numeric.py

```diff
@@ -1107,6 +1107,11 @@ def test_array_method(self):
 
         assert_equal(m.nonzero(), tgt)
 
+    def test_nonzero_invalid_object(self):
+        # gh-9295
+        a = np.array([np.array([1, 2]), 3])
+        assert_raises(ValueError, np.nonzero, a)
+
```

Comment:
Another useful test might be
```python
class BoolErrors:
    def __bool__(self):
        raise ValueError("Not allowed")

np.array([BoolErrors(), BoolErrors()]).nonzero()
```
which is similar to the test that came up for sorting objects with erroring comparators